tendermint/src/lite/error.rs

```diff
@@ -78,6 +78,10 @@ pub enum Kind {
     /// the implementation, if any.
     #[error("Implementation specific error")]
     ImplementationSpecific,
+
+    /// This is returned when a validator has signed but isn't a part of the validator set
+    #[error("Found faulty validator: {address}")]
+    FaultyValidator { address: account::Id },
```

Comment:
A way around this, could be to remove the address / account::Id and annotate the error in the implementation using the `fail!` macro. Or we could make use of `ImplementationSpecific` with the proper annotation instead (as it is in fact implementation specific). Not sure what would be the best approach here. Will `FaultyValidator` be used by the light module itself? If not, that could be seen as indicator that we do not necessarily need to define a particular error kind in the light module (currently still called lite) for this 🤔

Reply:
Note that if we decide to keep this error kind here, it should probably also be in the spec (cc @milosevic).

Reply:
Good point @Liamsi . Should `FaultyValidator` go under Tendermint module, then? Also, this can be converted into a generalized error type `FaultyFullNode` and this error should trigger verification logic to mark this as a "faulty node" in the address book as explained in the [spec](https://github.com/tendermint/spec/tree/master/spec/consensus/light-client#components) (if i'm reading it right). You could also attach a reason with this error, for example in this case it can be - 'the signed header received from this node contains commit from a validator not in the validator set' for more clarity. This way, we also get rid of the extra dependency. The reason why I think we shouldn't use `ImplementationSpecific` here is that this is a serious misbehaviour and must be distinctly identifiable. 

If that's not sth we want to do, I'm okay to use `ImplementationSpecific` but then I'm not sure how updating the address book would work.

Reply:
Thanks for the explanation! You're right:
> Should FaultyValidator go under Tendermint module, then?

I think that makes the most sense.

Reply:
Great! Thanks :)

Reply:
Unless the logic that that needs to handle the error kind `FaultyFullNode` will live inside light module (which it sounds like in the spec/t[he diagram](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/assets/light-node-image.png)). But then we do not want that additional dependency on `account::Id` (there are probably alternatives around this, e.g. as you mentioned above). If that is the simplest way to unblock you and this PR, we can also keep it here (and just remove the dependency on account::Id). It wouldn't be a biggie to move that error kind around later anyways.

Reply:
If needed in the verification logic directly, we might introduce a trait for testing equality between peers later (doesn't need to be the concrete account::Id).